### PR TITLE
Clq api fixes

### PR DESCRIPTION
**Pull request by rdingwell**
This PR contains fixes for issues related to changes in CLQ outcomes service implementation.  Specially, changes were made to race and gender handling along with the addition of added ethnicity to the API.  The gender changes are related to switching to SNOMEDCT codes instead of the hl7 fhir valuesets.  Race codes are restricted to a subset of 5 codes from within the hl7 race code valueset hierarchy.   

Additional changes are related to handling null pointers in tumor markers 

## Submitter:

**Running the Application**

- [ ] Manually tested in Chrome (primary browser for testing)
- [ ] Manually tested in IE/Safari (verify app loads and basic feature testing)
- [ ] Manually tested in Chrome in iPad mode (at minimum, viewing patient record)

**Task**

- [ ] Branch implements task as expected
- [ ] Key application features are updated as needed for task and tested (see: [Key Application Features](https://github.com/FluxNotes/flux/wiki/Key-Application-Features))
- [ ] Task specific definition of done (as described in the Jira task) is met, if applicable.

**Documentation**

- [ ] This pull request describes why these changes were made
- [ ] Recognizes any potential shortcomings/bugs in the description 
- [ ] Pre release script is updated on the [Pre Release Testing Script wiki page](https://github.com/FluxNotes/flux/wiki/Pre-Release-Testing-Script) (add/remove single features to test, do not run through full script)
- [ ] Update the list of key features with new features on the [Key Application Features wiki page](https://github.com/FluxNotes/flux/wiki/Key-Application-Features)
- [ ] Document any new APIs/extension points
- [ ] Additional technical components and libraries are documented and added to [wiki](https://github.com/FluxNotes/flux/wiki/About-Flux-Notes) as needed

**Code Quality**

- [ ] Code style guide is followed (see: [Code Style Guide](https://github.com/FluxNotes/flux/wiki/JavaScript-and-HTML-Code-Style-Guide))
- [ ] Code is commented

**Tests**

- [ ] Existing tests passed
- [ ] Added backend tests for new functionality added

## Reviewers:

- Check that code is maintainable and reusable. Code reuses existing code and infrastructure where appropriate.
- Check the PR and code accomplishes the task's purpose.
- Check that new tests appropriately test the new code, including edge cases. Other existing tests pass.
- Verify Pull Request checklist is correctly completed by submitter.
- Try to break the code

### CODE REVIEW

File: src/mcode-pilot/services/outcomes/CLQOutcomesService.js [hunk 1]
```diff
@@ -11,20 +13,60 @@ export default class CLQOutcomesService extends IOutcomesService {
...
+            displayName: display,
+            codeSystem: 'SNOMEDCT'
+        };
+    }
+
+    __raceCodeMapping(race) {
+        let code = find_race_code(race);
+        if (code) {
+            // if the value is one of the codes from the code system make sure it is one that clq supports
+            if (!['2028-9', '2106-3', '2054-5', '2131-1'].indexOf(code.code)) {
```
Comment (dehall) [inline src/mcode-pilot/services/outcomes/CLQOutcomesService.js]:
```suggestion
            if (!['2028-9', '2106-3', '2054-5', '2131-1'].includes(code.code)) {
```
`indexOf` returns `-1` if the given code isn't in the list, which is a truthy value, and `0` if the given code is the first item in the list ('2028-9') which is a falsy value. I think you want `includes` here

File: src/mcode-pilot/services/outcomes/CLQOutcomesService.js [hunk 2]
```diff
@@ -11,20 +13,60 @@ export default class CLQOutcomesService extends IOutcomesService {
         this.filters = params.filters;
     }
 
+    _genderMapping(gender) {
+        let lowered = gender ? _.toLower(gender) : null;
+        let code = 'UNK';
+        let display = 'UNKOWN';
```
Comment (dehall) [inline src/mcode-pilot/services/outcomes/CLQOutcomesService.js]:
```suggestion
        let display = 'UNKNOWN';
```

File: src/model/oncocore/FluxTumorMarker.js [hunk 3]
```diff
@@ -60,14 +60,14 @@ class FluxTumorMarker extends FluxEntry {
...
-        if (!this._tumorMarker.findingTopicCode || !this._tumorMarker.findingTopicCode.value) return null;
+        if (!(this._tumorMarker.findingTopicCode && this._tumorMarker.findingTopicCode.value && this._tumorMarker.findingTopicCode.value.coding && this._tumorMarker.findingTopicCode.value.coding.length > 0)) return null;
         return this._tumorMarker.findingTopicCode.value.coding[0];
     }
     
     // Returns a human-readable display text string
     get receptorType() { 
-        if (!this._tumorMarker.findingTopicCode || !this._tumorMarker.findingTopicCode.value) return null;
-        return this._tumorMarker.findingTopicCode.value.coding[0].displayText.value;
+        if (!this.receptorTypeCodeableConcept) return null;
```
Comment (dehall) [inline src/model/oncocore/FluxTumorMarker.js]: These changes here are going to be overwritten shortly by any changes in the `mcode_09` branch. Would it be possible to use the 09 branch for your testing?

Comment (rdingwell) [reply src/model/oncocore/FluxTumorMarker.js]: If the mcode_09 branch is going to be merged into master soon then this PR can wait until that time.

File: src/mcode-pilot/services/outcomes/race_codes.js [hunk 4]
```diff
@@ -0,0 +1,942 @@
...
+    {"level": "3","code": "2095-8","text": "Saipanese"},
+    {"level": "3","code": "2096-6","text": "Kiribati"},
+    {"level": "3","code": "2097-4","text": "Chuukese"},
+    {"level": "3","code": "2098-2","text": "Yapese"},
+    {"level": "2","code": "2100-6","text": "Melanesian"},
+    {"level": "3","code": "2101-4","text": "Fijian"},
+    {"level": "3","code": "2102-2","text": "Papua New Guinean"},
+    {"level": "3","code": "2103-0","text": "Solomon Islander"},
+    {"level": "3","code": "2104-8","text": "New Hebrides"},
+    {"level": "2","code": "2500-7","text": "Other Pacific Islander Note that this term remains in the table for completeness, even though within HL7, the notion of Other code is deprecated."},
```
Comment (dehall) [inline src/mcode-pilot/services/outcomes/race_codes.js]: Looks like a little extra got copy & pasted on this line.

Comment (rdingwell) [reply src/mcode-pilot/services/outcomes/race_codes.js]: Yeah, that was included in the list I copied the codes from.  Removed.

**Comment by jafeltra** [comment]
This PR looks good to me! Sorry I didn't realize I was the reviewer blocking this!

One unrelated thing I noticed is that when the side effects are not shown and you select a treatment to compare against, the side effects column appears again. I think that should be a separate Jira (as long as that's not intended).

**Pull request merged**